src/config/settings.rs

```diff
@@ -0,0 +1,123 @@
+/// A setting can be either a signed integer, unsigned integer, string, map or boolean.
+/// Maps could be created by using comma separated strings maybe
+#[derive(Clone, PartialEq, Debug)]
+pub enum Setting {
+    SInt(isize),
+    UInt(usize),
+    String(String),
+    Bool(bool),
+    Map(Vec<String>),
+}
+
+impl Setting {
+    // first element is the type:
+    //   b:true
+    //   i:-123
+    //   u:234
+    //   s:hello world
+    //   m:foo,bar,baz
+
+    /// Converts a string to a setting or None when the string is invalid
+    pub(crate) fn from_string(p0: &str) -> Option<Setting> {

```

Comment:
A setting can have different types. bools, ints, strings etc. This tries to accomodate that